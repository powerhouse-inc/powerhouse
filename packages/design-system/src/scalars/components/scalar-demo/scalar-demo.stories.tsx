import type { Meta, StoryObj } from "@storybook/react";
import { ScalarDemo } from "./scalar-demo";

const meta: Meta<typeof ScalarDemo> = {
  title: "Document Engineering/Complex Components/ScalarDemo",
  component: ScalarDemo,
  argTypes: {
    name: {
      control: {
        type: "text",
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    name: "Hello, World!",
  },
};
