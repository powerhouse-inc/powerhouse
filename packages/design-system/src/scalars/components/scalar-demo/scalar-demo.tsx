import { EmailAddress as EmailAddressScalar } from "@powerhousedao/scalars";

export interface ScalarDemoProps {
  name: string;
}

export const ScalarDemo: React.FC<ScalarDemoProps> = ({ name }) => {
  const result = EmailAddressScalar.schema.safeParse(name);
  console.log(result);

  return (
    <div className="bg-yellow-200 rounded-md p-3">Scalar Demo: {name}</div>
  );
};
