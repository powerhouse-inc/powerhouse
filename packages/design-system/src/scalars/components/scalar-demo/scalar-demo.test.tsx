import { render } from "@testing-library/react";

import { ScalarDemo } from "./scalar-demo";

describe("ScalarDemo Component", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(<ScalarDemo name="test" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
