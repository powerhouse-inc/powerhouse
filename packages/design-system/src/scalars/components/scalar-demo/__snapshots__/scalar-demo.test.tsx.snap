// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ScalarDemo Component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-yellow-200 rounded-md p-3"
  >
    Scalar Demo: test
  </div>
</DocumentFragment>
`;
